corehq/apps/repeaters/models.py

```diff
@@ -593,6 +593,10 @@ def _fail(self, reason, response):
             'target_url': self.url,
         })
 
+    def add_failure_reason(self, failure_reason):
```

Comment:
there's already the `_fail` method right above this that does what you are doing here, and also sets up whether to retry the repeater or not. I don't think you should need this. 
In theory, this property is already set when the `RepeatRecord` encounters a failure. Or is the way that Nikshay is returning the failure message not able to be saved that way?